Modules/mathmodule.c

```diff
@@ -2490,6 +2490,151 @@ math_isclose_impl(PyObject *module, double a, double b, double rel_tol,
 }
 
 
+/*[clinic input]
+math.comb
+
+    n: object
+
+    k: object
+    /
+
+Return the binomial coefficient indexed by the pair of integers n >= k >= 0.
+
+It is the coefficient of kth term in polynomial expansion of the expression
+(1 + x)^n. It is also known as the number of ways to choose an unordered
+subset of k elements from a fixed set of n elements, usually called
+*n choose k*.
+
+Raises a TypeError if argument(s) are non-integer and ValueError
+if argument(s) are negative or k > n.
+
+[clinic start generated code]*/
+
+static PyObject *
+math_comb_impl(PyObject *module, PyObject *n, PyObject *k)
+/*[clinic end generated code: output=bd2cec8d854f3493 input=75e1a19623bae7dc]*/
+{
+    if (!(PyLong_Check(n) && PyLong_Check(k))) {
+        PyErr_SetString(PyExc_TypeError,
+            "comb() only accepts integer arguments");
+        return NULL;
+    }
+    n = PyNumber_Long(n);
```

Comment:
If possible, move these checks and conversions into the argclinic definition:

```
    n: long
    k: long
```

Reply:
I left them as objects so that comb(10**200, 10**10) etc don't cause OverflowError.